gtk/docs.md

```diff
@@ -1 +1,68 @@
-<!-- file * -->
+<!-- struct ImageMenuItem -->
+`[Deprecated since 3.10]` A `ImageMenuItem` is a menu item which has an icon next to the text label.
+
+This is functionally equivalent to:
+
+
+```C
+  GtkWidget *box = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 6);
+  GtkWidget *icon = gtk_image_new_from_icon_name ("folder-music-symbolic", GTK_ICON_SIZE_MENU);
+  GtkWidget *label = gtk_label_new ("Music");
+  GtkWidget *menu_item = gtk_menu_item_new ();
+
+  gtk_container_add (GTK_CONTAINER (box), icon);
+  gtk_container_add (GTK_CONTAINER (box), label);
+
+  gtk_container_add (GTK_CONTAINER (menu_item), box);
+
+  gtk_widget_show_all (menu_item);
+```
+
+Note that the user may disable display of menu icons using
+the `Settings:gtk-menu-images` setting, so make sure to still
+fill in the text label. If you want to ensure that your menu items
+show an icon you are strongly encouraged to use a `MenuItem`
+with a `Image` instead.
+
+`ImageMenuItem` has been deprecated since GTK+ 3.10. If you want to
+display an icon in a menu item, you should use `MenuItem` and pack a
+`Box` with a `Image` and a `Label` instead. You should also consider
+using `Builder` and the XML `gio::Menu` description for creating menus, by
+following the [GMenu guide][https://developer.gnome.org/GMenu/]. You should
+consider using icons in menu items only sparingly, and for "objects" (or
+"nouns") elements only, like bookmarks, files, and links; "actions" (or
+"verbs") should not have icons.
+
+So before you did like this with ImageMenuItem:
+
+```Rust
+  let file_entry = ImageMenuItem::new_with_label("File");
+  let file_image = Image::new_from_file("some_image.png");
```

Comment:
This sounds like the opposite of minimizing modifications. If there's nothing wrong with the substance of that example, let's preserve it.